server/gitrest/packages/gitrest-base/src/routes/git/blobs.ts

```diff
@@ -6,17 +6,21 @@
 import { ICreateBlobParams } from "@fluidframework/gitresources";
 import { Router } from "express";
 import nconf from "nconf";
-import { IRepositoryManagerFactory } from "../../utils";
+import { Constants, IRepositoryManagerFactory } from "../../utils";
 import { handleResponse } from "../utils";
 
 export function create(store: nconf.Provider, repoManagerFactory: IRepositoryManagerFactory): Router {
     const router: Router = Router();
 
     router.post("/repos/:owner/:repo/git/blobs", async (request, response, next) => {
-        const resultP = repoManagerFactory.open(
-            request.params.owner,
-            request.params.repo,
-        ).then(async (repoManager) => repoManager.createBlob(
+        const storageName: string | undefined = request.get(Constants.StorageNameHeader);
```

Comment:
Good idea :) do you think this would also help for the `storage-routing-id` param that we have in summaries.ts? Since we only use `storage-routing-id` in summaries.ts, I was wondering if it would be better to keep those two (`storage-name`) apart

Reply:
Awesome! Looks great, thanks!

> do you think this would also help for the storage-routing-id param that we have in summaries.ts? Since we only use storage-routing-id in summaries.ts, I was wondering if it would be better to keep those two (storage-name) apart

I agree with how it is now. Leaving storage-routing-id as a summaries-specific header is fine. I think when you make your Telemetry changes to Gitrest though, you might want to consider parsing it in all routes (only throwing if invalid for summaries routes). The reason being that Historian already sends the header in every request anyways, and it will be helpful to have tenantId and docId for telemetry :)